tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmokeTestsParser.java

```diff
@@ -0,0 +1,362 @@
+        forEach(operationShape -> {
+            SmokeTestsTrait smokeTestsTrait = operationShape.getTrait(SmokeTestsTrait.class).get();
+            //get serviceShape
+            String serviceName = operationToServiceMap.get(operationShape.getId());
+            ServiceShape serviceShape = serviceShapeMap.get(serviceName);
+            //System.out.println("OperationShape: " + operationShape.getId().getName());
+            //System.out.println("serviceName: " + serviceName);
+
+            List<SmokeTestData> tests = parseSmokeTests(
+                smokeTestsTrait, 
+                operationShape,
+                serviceShape );
+            //add to tests for the same service
+            if(serviceSmokeTestsMap.containsKey(serviceShape))
+            {
+                serviceSmokeTestsMap.get(serviceShape).addAll(tests);
+            }
+            else
+            {
+                serviceSmokeTestsMap.put(serviceShape, tests);
+            }
+
+        });
+        
+        return serviceSmokeTestsMap;
+    }
+
+    @Override
+    public void run(){
+        //System.out.println("run smoke tests parser");
+        SmokeTestsSourceDelegator delegator = new SmokeTestsSourceDelegator(this.context.getFileManifest(), this.symbolProvider);
+        SmokeTestsCMakeDelegator cmakedelegator = new SmokeTestsCMakeDelegator(this.context.getFileManifest(), this.symbolProvider);
+        Map<ServiceShape, List<SmokeTestData> > smoketests =  extractServiceSmokeTests();
+
+        //make service specific folder
+        smoketests.entrySet().stream().forEach(entry -> {
+                ServiceShape serviceShape = entry.getKey();
+                try{
+                    String client = getServiceName(serviceShape);
+                    String c2jClientname = SmithyC2JNamespaceMap.getC2JServiceName(toKebabCase(client));
+                    
+                    Path relativePath = Paths.get( c2jClientname );
+                    System.out.println(toKebabCase(client) + " mapped to " + c2jClientname);
+                    
+                    delegator.useFileWriter( relativePath.toString() + "/"+ removeSpaces(client) + "SmokeTests.cpp", client, writer -> {
+                        System.out.println(String.format("generating smoke test source code=%s",relativePath.toString() + "/"+ removeSpaces(client) + "SmokeTests.cpp"));
+                        writer.generate(entry.getValue());              
+                    });
+
+                    cmakedelegator.useFileWriter( relativePath.toString() + "/"+ "CMakeLists.txt", c2jClientname, writer -> {
+                        System.out.println(String.format("generating smoke test cmake code=%s",relativePath.toString() + "/"+ "CMakeLists.txt"));
+                        writer.generate();
+                    });
+                }
+                catch (Exception e) {
+                    System.err.println("Exception detected=" + e.toString());
```

Comment:
remove catch or re-throw runtime exception so the generation will actually fail

Reply:
ack